fix restore for postgres 13

risicle opened the pull request:
This PR was originally just to add test coverage for postgres 13:

> Still trying to limit the number of tests as they are very slow, make sure we at least test the versions of postgres at the limits of our support - the oldest (currently 10) and the newest (currently 13).

> This does mean that 11 and 12 technically will get less coverage.

risicle committed:
blackbox tests: use more appropriate coverage of postgres versions

still trying to limit the number of tests as they are very slow, make
sure we at least test the versions of postgres at the limits of our
support - the oldest (currently 10) and the newest (currently 13).

this does mean that 11 and 12 technically will get less coverage.

schmie reviewed (approved):
:+1: Increasing/instating coverage for PostgreSQL version 13 is clearly needed - if the overall runtime requires to removes things, I agree that dropping tests "in the middle" makes the most sense.
I think this could do with another pair of (TL) eyes, though @paroxp :)

paroxp reviewed (commented):
re:@schmie In theory, this is fine.

 I'm not big on limiting tests for 11 or 12 - it's still our offering, we need to be assured they're falling under the same set of standards...

But I cannot think of a nice way of doing it. Besides, nothing is set in stone. I suppose we could come back to it at some point and figure out a different mechanism to address all the plans reliably and quickly...

schmie committed:
Exlude `rdstopmgr` from DROP USER in ResetState

The user role `rdstopmgr` was introduced by AWS in RDS PostgreSQL 13.
This user cannot be dropped, but is also not a superuser. This caused
restores from snapshots to break as the operation includes a ResetState.
Thus we had to include a special case in the listNonSuperUsers function.
